### ISSUE TITLE
Running nested queries fails?

### ISSUE THREAD

**Issue by Script47** [dfahlander/Dexie.js#522]
Take the following code,

```javascript
    db_class.user.get(function (user) {
        if (!user.is_drag_and_drop_enabled) {
            db_class.user.update({
                is_drag_and_drop_enabled: 0
            }, function (updated) {});
        }
    });
```

when it executes I get this error in the console and it does not work, I get the following error,

`dexie.js:1 Unhandled rejection: ReadOnlyError: Transaction is readonly`

why is this?

**Comment by dfahlander** [comment]
I assume you have your code surrounded by a transaction scope (inline or in a caller of your function) :

```js
db.transaction('r',... () => {
    //your code here
}) ;
```

**Comment by Script47** [comment]
@dfahlander Out of curiosity, why is it like that?

**Comment by dfahlander** [comment]
It's becuase the indexedDB transaction was opened in a readonly state. For a full explanation of how transaction scopes work, read throw this doc page. I hope it explains the details: http://dexie.org/docs/Dexie/Dexie.transaction()

**Comment by lddubeau** [comment]
There is one thing that may trip people who are unaware of the details of how Dexie works. If they start a query that reads a table and during this query, start another query that modifies the database, they may run into `ReadOnlyError: Transaction is readonly`. 

I ran into the issue with code that did a `table.where(...).each(...)` and the callback to `.each()` was changing the database. My immediate thought was "hmm.... I'm not using transactions" because I never call `db.transaction` *explicitly*, but the outer query does start a readonly transaction *implicitly*. And the code in `.each()` executes within that implicit transaction.

**Comment by dfahlander** [comment]
That's true. The issue is there's no way in indexedDB to upgrade a transaction to be readwrite. So there would be two options of solving this. Either do the things within each as separate transactions (which could  lead to performance issues), or better, provide a more explaining error message when each() is used transactionless. (Suggest using [Collection.modify()](http://dexie.org/docs/Collection/Collection.modify()) instead, or use a 'rw' transaction scope around Collection.each())

**Comment by wearetelescopic** [comment]
> There is one thing that may trip people who are unaware of the details of how Dexie works. If they start a query that reads a table and during this query, start another query that modifies the database, they may run into `ReadOnlyError: Transaction is readonly`.
> 
> I ran into the issue with code that did a `table.where(...).each(...)` and the callback to `.each()` was changing the database. My immediate thought was "hmm.... I'm not using transactions" because I never call `db.transaction` _explicitly_, but the outer query does start a readonly transaction _implicitly_. And the code in `.each()` executes within that implicit transaction.

If you run into this issue but there's no easy way round, because you need to some actions WITHIN the .each loop - I can recommend wrapping your actions into a timeout.
